### PR TITLE
Full Auto Update: no crit firing, no spamclicking, more autofire, more balance

### LINKED ISSUE #57963: Full-auto weapons can be fired while in crit
## Reproduction: 
1. Get L6/Tesla Cannon/laser minigun, and an antidrop implant.
2. Get put into crit for whatever reason
3. Gun down your attacker while they scream.
4. For extra salt, if you die while firing the full-auto weapon, you can continue firing while dead until you run out of ammunition.

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

You can no longer fire full-auto while in crit (Fixes #57963)
You can no longer spamclick to fire infinitely fast
Adds full-auto to saber smg (0.2 seconds), borg l6 (0.3 seconds), tommy gun (0.1 seconds), security autorifle (0.3 seconds)
tesla cannon projectiles now cost less to shoot and are tesla level zaps so they should now work against swarmers for example, and a full mag will actually crit a person now that they cost less
l6 projectiles other than hollow point now do less damage

## Why It's Good For The Game

adds some really needed fixes and balance for fullauto

## Changelog
:cl:
balance: adds fullauto to saber smg's, borg l6's, tommy guns and security autorifles
balance: tesla cannon sucks less
balance: l6 bullets now do less damage
fix: fixes full auto letting you fire in crit or spamclick
/:cl:

### CODE REVIEW

**Comment by rdragan** [comment]
What do the Swedish call this? A smorgasbord right?

**Comment by Livrah** [comment]
NO FULL AUTO IN BUILDING!
Why no auto for non borg L6.

**Comment by Fikou** [comment]
> NO FULL AUTO IN BUILDING!
> Why no auto for non borg L6.

what are u talking about it got full auto like a month ago

**Comment by necromanceranne** [comment]
You should give full auto to the plastikov and APS. I desire to spray the walls and not hit anyone at all before emptying my magazine immediately, as god intended.

File: code/datums/components/fullauto.dm [hunk 1]
```diff
@@ -261,6 +258,8 @@
 
 /obj/item/gun/proc/do_autofire(datum/source, atom/target, mob/living/shooter, params)
 	SIGNAL_HANDLER
+	if(semicd || shooter.stat || !can_shoot())
```
Comment (Fikou) [inline code/datums/components/fullauto.dm:261]:
```suggestion
	if(semicd || shooter.stat)
```

**Comment by ATH1909** [comment]
the L6 no longer 3 hit crits... sad!

does anything else use L6 bullets? if so, you should add a damage multiplier to the L6 itself instead of modifying the bullet's damage

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -63,6 +48,26 @@
...
+	burst_size = 1
+	actions_types = list()
+	mag_display = TRUE
+	empty_indicator = TRUE
+	mag_type = /obj/item/ammo_box/magazine/smgm9mm
+	pin = null
+	bolt_type = BOLT_TYPE_LOCKING
+	show_bolt_icon = FALSE
+
+/obj/item/gun/ballistic/automatic/proto/ComponentInitialize()
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:64]: Don't use ComponentInitialize, it's deprecated. Just use Initialize.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 3]
```diff
@@ -110,6 +115,10 @@
 	mag_display_ammo = TRUE
 	empty_indicator = TRUE
 
+/obj/item/gun/ballistic/automatic/wt550/ComponentInitialize()
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:118]: Same

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -214,12 +223,17 @@
...
 	can_suppress = FALSE
-	burst_size = 4
+	burst_size = 1
+	actions_types = list()
 	fire_delay = 1
 	bolt_type = BOLT_TYPE_OPEN
 	empty_indicator = TRUE
 	show_bolt_icon = FALSE
 
+/obj/item/gun/ballistic/automatic/tommygun/ComponentInitialize()
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:233]: Same

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**